state-chain/pallets/cf-ingress-egress/src/tests.rs

```diff
@@ -1474,3 +1474,28 @@ fn consolidation_tx_gets_broadcasted_on_finalize() {
 		));
 	});
 }
+
+#[test]
+fn all_batch_errors_are_logged_as_event() {
+	new_test_ext().execute_with(|| {
+		ScheduledEgressFetchOrTransfer::<Test>::set(vec![FetchOrTransfer::<Ethereum>::Transfer {
+			asset: ETH_ETH,
+			amount: 1_000,
+			destination_address: ALICE_ETH_ADDRESS,
+			egress_id: (ForeignChain::Ethereum, 1),
+		}]);
+		MockEthAllBatch::set_success(false);
+
+		// Expect the egress to fail
+		matches!(
+			IngressEgress::do_egress_scheduled_fetch_transfer(),
+			sp_runtime::TransactionOutcome::Rollback(..)
+		);
+
+		System::assert_last_event(RuntimeEvent::IngressEgress(
+			crate::Event::<Test>::FailedToBuildAllBatchCall {
+				error: cf_chains::AllBatchError::Other,
```

Comment:
I think this test passes because calling `do_egress_scheduled_fetch_transfer` outside of a transactional wrapper *doesn't* roll back the storage. 

A better approach might be to refactor `do_egress_scheduled_fetch_transfer` so that it returns a Result and simply decorate the fn with `#[transactional]`.